Sources/ArgumentParserToolInfo/ToolInfo.swift

```diff
@@ -42,6 +42,69 @@ public struct ToolInfoV0: Codable, Hashable {
 /// All information about a particular command, including arguments and
 /// subcommands.
 public struct CommandInfoV0: Codable, Hashable {
+  public enum DiscussionV0: Codable, Hashable {
+    case staticText(String)
+    case enumerated([ValueV0])
+
+    public struct ValueV0: Codable, Hashable {
+      var name: String
+      var discussion: String
+
+      public init(name: String, discussion: String) {
+        self.name = name
+        self.discussion = discussion
+      }
+    }
+
+    public init(_ text: String) {
+      self = .staticText(text)
+    }
+
+    public init(_ values: [ValueV0]) {
+      self = .enumerated(values)
+    }
+
+    enum CodingKeys: String, CodingKey {
+      case enumerated = "command_values"
+      case staticText = "command_text"
+    }
+
+    public func encode(to encoder: any Encoder) throws {
+      var container = encoder.singleValueContainer()
+      switch self {
+      case .staticText(let s):
+        try container.encode(s)
+      case .enumerated(let s):
+        try container.encode(s)
+      }
+    }
+
+    public init(from decoder: any Decoder) throws {
+      let container = try decoder.singleValueContainer()
+      if let value = try? container.decode(String.self) {
+        self = .staticText(value)
+        return
+      }
+      if let values = try? container.decode([ValueV0].self) {
+        self = .enumerated(values)
+        return
+      }
+      throw DecodingError.typeMismatch(DiscussionV0.self, DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Unexpected assosiated type for DiscussionV0."))
+    }
+
+    public var isEmpty: Bool {
```

Comment:
Yeah, I think this can get redundant and it's probably better to just have Discussion properties be optional - this way I can just check if the property exists or not and assert that it contents are not empty on creation.